app/models/drupal_node_revision.rb

```diff
@@ -43,6 +43,13 @@ def inline_tags
     end
   end
 
+  # search for inline hashtags(such as #hashtag) and create a new tag
+  def inline_hashtags
+    self.body.scan(/[^\n]\#([\w-]+)/).each do |match|
```

Comment:
Actually headers in strict markdown require a space after the hash, but as
many people (and even our templates) don't follow that, we've made
single-hash headers read as hashtags. This is a compromise as there have
been very few single-hash headers on the site since they're not provided in
templates.

I believe we have a set of tests for this in comments, Ananyo - could you
link chongr to the relevant tests?
On Jul 12, 2016 10:20 AM, "chongr" <notifications@github.com> wrote:

> In app/models/drupal_node_revision.rb
> <https://github.com/publiclab/plots2/pull/609#discussion_r70481096>:
>
> > @@ -43,6 +43,13 @@ def inline_tags
> >      end
> >    end
> >
> > +  # search for inline hashtags(such as #hashtag) and create a new tag
> > +  def inline_hashtags
> > +    self.body.scan(/[^\n]\#([\w-]+)/).each do |match|
>
> That makes sense, but I think that regex also doesn't work for things such
> as this:
>
> #Heading 1
> Here is a #hashtag
> #Heading 2
> Here is another #hashtag
>
> that regex will match heading 2 if I'm not mistaken. Let me know if you
> have an idea of how to avoid markdown headers. Thanks!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/609/files/873d228c866b682ad8c9f118b9691a9b77bf6855#r70481096>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABfJ68ZUt2KGcWeGToEPD97FZLR1jqkks5qU8zNgaJpZM4JKHF4>
> .
>